packages/material-ui-styled-engine-sc/src/StylesProvider/StylesProvider.js

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { StyleSheetManager } from 'styled-components';
+
+let injectFirstNode;
+
+export function StylesProvider(props) {
+  const { injectFirst, children } = props;
+
+  React.useEffect(() => {
+    if (injectFirst && !injectFirstNode) {
+      const head = document.head;
+      injectFirstNode = document.createElement('style');
+      injectFirstNode.setAttribute('data-styled', 'active');
+      head.insertBefore(injectFirstNode, head.firstChild);
+    }
```

Comment:
Should we clean up the node?

Reply:
Done with https://github.com/mui-org/material-ui/pull/23934/commits/d1e122da030026817312a1dda8556d5866d3a334

Reply:
Let me know if you think something else needs to be added/changed